packages/aws-cdk-lib/aws-ecs/lib/amis.ts

```diff
@@ -152,13 +160,14 @@ export class EcsOptimizedAmi implements ec2.IMachineImage {
       + (this.windowsVersion ? 'image_id' : 'recommended/image_id');
 
     this.cachedInContext = props?.cachedInContext ?? false;
+    this.additionalCacheKey = props?.additionalCacheKey;
```

Comment:
We should probably add some validation to say that `additionalCacheKey` only makes sense if `cachedInContext==true` ?